dali/operators/generic/one_hot.cc

```diff
@@ -74,13 +78,33 @@ TensorShape<> determine_shape(TensorShape<> in_shape, int num_classes, int axis,
   axis = axis < 0 ? in_shape.sample_dim() : axis;
   int outer_axes = axis;
   int inner_axes = in_shape.sample_dim() - axis;
-  auto outer =  in_shape.first(outer_axes);
+  auto outer = in_shape.first(outer_axes);
   auto inner = in_shape.last(inner_axes);
   return shape_cat(shape_cat(outer, num_classes), inner);
 }
 
 }  // namespace
 
+TensorLayout OneHot::GetOutputLayout(const HostWorkspace &ws, int placement_axis) const {
+  const auto &input = ws.template InputRef<CPUBackend>(0);
+  const auto &output = ws.template OutputRef<CPUBackend>(0);
+  auto in_layout = input.GetLayout();
+  auto in_layout_size = in_layout.size();
+  auto out_sample_dim = output.shape().sample_dim();
```

Comment:
Maybe it should accept `out_sample_dim` as an argument, otherwise someone may call this function to the place where output.shape() is not defined yet.

Reply:
OK, added it as a parameter.